uniffi_aries_vcx/demo/app/src/main/java/org/hyperledger/ariesvcx/ScanScreen.kt

```diff
@@ -0,0 +1,132 @@
+import android.util.Base64
+@Composable
+fun ScanScreen() {
+    var code by remember {
+        mutableStateOf("")
+    }
+
+    val context = LocalContext.current
+    val lifecycleOwner = LocalLifecycleOwner.current
+    val cameraProviderFuture = remember {
+        ProcessCameraProvider.getInstance(context)
+    }
+    var showDialog by remember { mutableStateOf(false) }
+
+    if (showDialog) {
+        val encoded = Uri.parse(code)?.getQueryParameter("c_i")
+        val decoded =  Base64.decode(encoded, Base64.DEFAULT).toString()
+        AlertDialog(
+            onDismissRequest = { showDialog = false },
+            title = { Text("Accept this invitation?") },
+            text = { Text(decoded) },
+            confirmButton = {
+                TextButton(onClick = {
+
+                }) {
+                    Text("Accept")
+                }
+            },
+            dismissButton = {
+                TextButton(onClick = { showDialog = false }) {
+                    Text("Cancel")
+                }
+            },
+        )
+    }
+
+    var hasCamPermission by remember {
+        mutableStateOf(
+            ContextCompat.checkSelfPermission(
+                context,
+                Manifest.permission.CAMERA
+            ) == PackageManager.PERMISSION_GRANTED
+        )
+    }
+    val launcher = rememberLauncherForActivityResult(
+        contract = ActivityResultContracts.RequestPermission(),
+        onResult = { granted ->
+            hasCamPermission = granted
+        }
+    )
+    LaunchedEffect(key1 = true) {
+        launcher.launch(Manifest.permission.CAMERA)
+    }
+    Column(
+        modifier = Modifier.fillMaxSize().padding(50.dp),
+        verticalArrangement = Arrangement.Center,
+        horizontalAlignment = Alignment.CenterHorizontally,
+    ) {
+        if (hasCamPermission) {
+            AndroidView(
+                factory = { context ->
+                    val previewView = PreviewView(context)
+                    val preview = Preview.Builder().build()
+                    val selector = CameraSelector.Builder()
+                        .requireLensFacing(CameraSelector.LENS_FACING_BACK)
+                        .build()
+                    preview.setSurfaceProvider(previewView.surfaceProvider)
+                    val imageAnalysis = ImageAnalysis.Builder()
+                        .setTargetResolution(
+                            Size(
+                                320,
+                                480
+                            )
+                        )
+                        .setBackpressureStrategy(ImageAnalysis.STRATEGY_KEEP_ONLY_LATEST)
+                        .build()
+                    imageAnalysis.setAnalyzer(
+                        ContextCompat.getMainExecutor(context),
+                        QrCodeAnalyzer { result ->
+                            code = result
+                            showDialog = true
+                        }
```

Comment:
IMO it might be cleaner if we do the following:
1. rename `code` to something more specific, e.g. `scannedQRCodeText`
2. make `scannedQRCodeText` a nullable string
3. remove `showDialog`
4. instead of `if (showDialog) { ... }` you can do `scannedQRCodeText?.let { text -> ... }`
5. have the dismiss button have an action that sets `scannedQRCodeText = null`

just lets us remove a state variable, which is usually nice iMO